conn.go

```diff
@@ -178,9 +178,9 @@ func parseDataUnit(r io.Reader) (uint32, error) {
 	if err != nil {
 		return 0, err
 	}
-	n -= 4 // https://tools.ietf.org/html/rfc5734#section-4
-	if n < 0 {
+	if n < 4 {
 		return 0, io.ErrUnexpectedEOF
 	}
-	return n, err
+	// https://tools.ietf.org/html/rfc5734#section-4
+	return n - 4, err
```

Comment:
Isn't the bug just more hidden now, since the return type is `uint32` and we're still subtracting 4?

Reply:
Line 181 `if n < 4 {` returns early, short-circuiting the subtraction.

Reply:
👍